docs/contributors/block-assets.md

```diff
@@ -1 +1,56 @@
+# Block Script Assets
 
+[Block Types](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/src/BlockTypes) are often responsible for enqueing script assets that make blocks functional on both the front-end and within the editor. Additionally, some block scripts require extra data from the server and thus have extra dependencies that need to be loaded.
+
+For performance reasons the blocks plugin ensures assets and asset data is made available only as needed.
+
+## When are assets needed?
+
+Assets are needed when we know a block will being rendered. 
+
+In the context of [Block Types](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/src/BlockTypes), assets and asset data is enqueued within the block `render()` method.
+
+In an admin editor context we must also ensure asset *data* is available when the `enqueue_block_editor_assets` hook is fired. That is because block scripts are enqueued ready for the Block Inserter, but the block may not be rendered. 
+
+Note: `enqueue_block_editor_assets` fires regardless of whether or not a block has been rendered in the editor context, so unless handled correctly, block data may be loaded twice.
```

Comment:
Should mention be made here about _how_ to handle this correctly or that using the provided api takes care of that?